pkg/loghttp/push/push.go

```diff
@@ -58,7 +58,11 @@ type TenantsRetention interface {
 	RetentionPeriodFor(userID string, lbs labels.Labels) time.Duration
 }
 
-type RequestParser func(userID string, r *http.Request, tenantsRetention TenantsRetention) (*logproto.PushRequest, *Stats, error)
+type Limits interface {
+	OTLPConfig(userID string) OTLPConfig
+}
```

Comment:
Not sure why a new interface here is helpful? Can't we use the existing overrides?

Reply:
It makes it easier to write tests, and we can't reference overrides since that would cause circular dependency.